src/js/structs/Job.js

```diff
@@ -0,0 +1,21 @@
+import Item from './Item';
+
+module.exports = class Job extends Item {
+  constructor() {
+    super(...arguments);
+
+    const id = this.getId();
+    if (id !== '/' && (!id.startsWith('/') || id.endsWith('/'))) {
+      throw new Error(`Id (${id}) must start with a leading slash ("/") and should not end with a slash, except for root id which is only a slash`);
+    }
+  }
+
+  getId() {
+    return this.get('id') || '/';
+  }
+
+  getName() {
+    return this.getId().split('/').pop();
```

Comment:
@mlunoe 
My earlier comments pointed more in the direction, that `getName` should be able to handle `id`s with an ending slash, so that the correct name will be returned anyway.
This makes the application more fail safe. Now with the thrown exception, the opposite is the case.